thirdparty/raknet/RakNetSocket2_Windows_Linux.cpp

```diff
@@ -82,7 +82,12 @@ void GetMyIP_Windows_Linux_IPV4( SystemAddress addresses[MAXIMUM_NUMBER_OF_INTER
 
 	if ( phe == 0 )
 	{
+#if !defined(TARGET_OS_IPHONEOS) || TARGET_OS_IPHONEOS == 0
+        // iOS 6 SDK makes phones hate this for some reason
+        // gethostbyname didn't return NULL on iPhone 5 running iOS 6.1.4 on iOS 5 SDK,
+        // but did return NULL when running on the same hardware on iOS 6.1 SDK
 		RakAssert(phe!=0);
+#endif
```

Comment:
Modification of thirdparty libraries isn't recommended but official raknet repository is archived

Reply:
Maybe check `errno` if `gethostbyname` returns NULL?

Reply:
This will crash in debug mode due to an assertion if I did not modify it. There seemed to be no logical way for me to fix this.

Reply:
That `gethostbyname` returns NULL is an error. Check `errno` and print the error number (or convert it to a string with `strerror(errno)`).

Reply:
And what would the solution to something like this be?

Reply:
Look at the errno and see what's wrong? strerror will tell you the class of error you are encountering, and then you can cross reference it in the docs to determine _why_ `gethostbyname` is returning NULL.

Reply:
That is assuming it's just some minor problem with RakNet that I would have to fix. But realistically, how would I even begin to go about solving a problem with RakNet? I don't believe it's worth it. It just seems like an iOS skill issue.